pinot-query-runtime/src/test/resources/queries/TableExpressions.json

```diff
@@ -0,0 +1,93 @@
+{
+  "where_clause_tests": {
+    "psql": "7.2.2",
+    "tables": {
+      "tbl": {
+        "schema": [
+          {"name": "strCol", "type": "STRING"},
+          {"name": "intCol", "type": "INT"}
+        ],
+        "inputs": [
+          ["foo", 1],
+          ["bar", 2],
+          ["alice", 42],
+          ["bob", 196883]
+        ]
+      }
+    },
+    "queries": [
+      { "sql": "SELECT * FROM {tbl} WHERE intCol > 5" },
+      { "sql": "SELECT * FROM {tbl} WHERE strCol IN ('foo', 'bar')" },
+      { "sql": "SELECT * FROM {tbl} WHERE intCol IN (196883, 42)" },
+      { "sql": "SELECT * FROM {tbl} WHERE strCol IN (SELECT strCol FROM {tbl} WHERE intCol > 100)" },
+      { "sql": "SELECT * FROM {tbl} WHERE intCol < (SELECT SUM(intCol) FROM {tbl} AS b WHERE strCol BETWEEN 'bar' AND 'foo')" },
+      { "sql": "SELECT * FROM {tbl} WHERE intCol BETWEEN 0 AND 100 AND strCol BETWEEN 'bar' AND 'foo'" },
+      {
+        "ignored": true,
+        "comments": "Relation Decorrelator not supported",
+        "sql": "SELECT * FROM {tbl} AS a WHERE a.strCol IN (SELECT b.strCol FROM {tbl} AS b WHERE b.intCol = a.intCol + 1)"
```

Comment:
For subquery in `WHERE` with `IN`, can we also add `NOT IN` ?

Reply:
For `IN` clause subquery, let's also add test that the exact same query that uses `ANY` (supported by postgres) instead of `IN` returns identical results.

Reply:
Same goes for `SOME` I guess

Reply:
added. none are supported but added with ignore flag